numpy/core/tests/test_simd_module.py

```diff
@@ -0,0 +1,103 @@
+import pytest
+from numpy.core._simd import targets
+
+npyv = None
+npyv2 = None
+for target_name, npyv_mod in targets.items():
+    if npyv:
+        if npyv_mod and npyv_mod.simd:
+            npyv2 = npyv_mod
+            break
+        continue
+    if npyv_mod and npyv_mod.simd:
+        npyv = npyv_mod
+
+unsigned_sfx = ["u8", "u16", "u32", "u64"]
+signed_sfx = ["s8", "s16", "s32", "s64"]
+fp_sfx = ["f32"]
+if npyv and npyv.simd_f64:
+    fp_sfx.append("f64")
+
+int_sfx = unsigned_sfx + signed_sfx
+all_sfx = unsigned_sfx + int_sfx
+
+@pytest.mark.skipif(not npyv, reason="could not find any SIMD extension with NPYV support")
+class Test_SIMD_MODULE(object):
+    def test_num_lanes(self):
+        for sfx in all_sfx:
+            nlanes = getattr(npyv, "nlanes_" + sfx)
+            vector = getattr(npyv, "setall_" + sfx)(1)
+            assert len(vector) == nlanes
+
+    def test_type_name(self):
+        for sfx in all_sfx:
+            vector = getattr(npyv, "setall_" + sfx)(1)
+            assert vector.__name__ == "npyv_" + sfx
+
+    def test_raises(self):
+        def assert_raises(e, callback, *args):
+            __tracebackhide__ = True  # Hide traceback for py.test
+            try:
+                callback(*args)
+                raise AssertionError("expected to raise " + e.__name__)
+            except e:
+                pass
+
+        a, b = [npyv.setall_u32(1)]*2
+        for sfx in all_sfx:
+            vcb = lambda intrin: getattr(npyv, f"{intrin}_{sfx}")
+            assert_raises(TypeError, vcb("add"), a)
```

Comment:
No need to build your own `assert_raises` here
```suggestion
        a, b = [npyv.setall_u32(1)]*2
        for sfx in all_sfx:
            vcb = lambda intrin: getattr(npyv, f"{intrin}_{sfx}")
            pytest.raises(TypeError, vcb("add"), a)
```